commit b1af26c245545a289b331c7b71996ecd88321540
Author: Quinn Tran <qutran@marvell.com>
Date:   Tue Oct 26 04:54:05 2021 -0700

    scsi: qla2xxx: edif: Flush stale events and msgs on session down
    
    On session down, driver will flush all stale messages and doorbell
    events. This prevents authentication application from having to process
    stale data.
    
    Link: https://lore.kernel.org/r/20211026115412.27691-7-njavali@marvell.com
    Fixes: 4de067e5df12 ("scsi: qla2xxx: edif: Add N2N support for EDIF")
    Reviewed-by: Himanshu Madhani <himanshu.madhani@oracle.com>
    Co-developed-by: Karunakara Merugu <kmerugu@marvell.com>
    Signed-off-by: Karunakara Merugu <kmerugu@marvell.com>
    Signed-off-by: Quinn Tran <qutran@marvell.com>
    Signed-off-by: Nilesh Javali <njavali@marvell.com>
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>

diff --git a/drivers/scsi/qla2xxx/qla_target.c b/drivers/scsi/qla2xxx/qla_target.c
index b3478ed9b12e..edc34e69d75b 100644
--- a/drivers/scsi/qla2xxx/qla_target.c
+++ b/drivers/scsi/qla2xxx/qla_target.c
@@ -1003,6 +1003,7 @@ void qlt_free_session_done(struct work_struct *work)
 					"%s bypassing release_all_sadb\n",
 					__func__);
 			}
+			qla_edif_clear_appdata(vha, sess);
 			qla_edif_sess_down(vha, sess);
 		}
 		qla2x00_mark_device_lost(vha, sess, 0);